### PR TITLE
feat: connect mask

### PR DESCRIPTION
**Pull request by yanzhihong23**
## Description

Closes # (NO_ISSUE)
1. Connect Mask 
2. Pin Extension Tip
3. Twitter User Guide

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Previews

image

image

image

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 1]
```diff
@@ -477,31 +413,54 @@ function SetupGuideUI(props: SetupGuideUIProps) {
...
     }
 
     const onClose = () => {
         currentSetupGuideStatus[ui.networkIdentifier].value = ''
     }
 
+    const onDone = () => {
+        // check user guide status
+        const network = ui.networkIdentifier
+        if (network === 'twitter.com' && userGuideStatus[network].value !== 'completed') {
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx:470]: what about other network?

Comment (yanzhihong23) [reply packages/mask/src/components/InjectedComponents/SetupGuide.tsx:470]: user guide is designed for twitter network currently.

**Comment by wenluomask** [comment]
Refer to https://mask.atlassian.net/jira/software/c/projects/MASK/boards/20?modal=detail&selectedIssue=MASK-294

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 2]
```diff
@@ -436,6 +283,75 @@ function FindUsername({ username, onConnect, onDone, onClose, onUsernameChange =
...
+    onDone?: () => void
+}
+function PinExtension({ onDone }: PinExtensionProps) {
+    const pinImg = new URL('../../resources/extensionPinned.png', import.meta.url).toString()
+    const { classes } = useWizardDialogStyles()
+    const { t } = useI18N()
+    const [checked, setChecked] = useState(true)
+
+    useEffect(() => {
+        dismissPinExtensionTip.value = checked
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx]: Why not change this value in the event handler below?

File: packages/icons/general/Verified.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { createIcon } from '../utils'
+import type { SvgIcon } from '@mui/material'
+
+export const VerifiedIcon: typeof SvgIcon = createIcon(
```
Comment (septs) [inline packages/icons/general/Verified.tsx:4]: plz run `pnpm -w lint`

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 4]
```diff
@@ -411,10 +415,9 @@ function SetupGuideUI(props: SetupGuideUIProps) {
         let reloaded = false
         const handler = () => {
             // twitter will redirect to home page after login
-            if (!reloaded && window.location.pathname === '/home') {
-                reloaded = true
-                window.location.reload()
-            }
+            if (!(!reloaded && window.location.pathname === '/home')) return
+            reloaded = true
+            window.location.reload()
```
Comment (septs) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx]:
```suggestion
            if (!(!reloaded && location.pathname === '/home')) return
            reloaded = true
            location.reload()
```

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 5]
```diff
@@ -460,51 +396,98 @@ function SetupGuideUI(props: SetupGuideUIProps) {
...
+        if (username || ui.networkIdentifier !== 'twitter.com') return
+        // In order to collect user info after login, need to reload twitter once
+        let reloaded = false
+        const handler = () => {
+            // twitter will redirect to home page after login
+            if (!(!reloaded && location.pathname === '/home')) return
+            reloaded = true
+            location.reload()
+        }
+        window.addEventListener('locationchange', handler)
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx:434]: FYI: packages/mask/src/utils/hooks/useLocationChange.ts

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 6]
```diff
@@ -460,51 +396,98 @@ function SetupGuideUI(props: SetupGuideUIProps) {
...
-    useEffect(
-        () =>
-            activatedSocialNetworkUI.collecting.identityProvider?.recognized.addListener((val) => {
-                if (username === '' && !val.identifier.isUnknown) setUsername(val.identifier.userId)
-            }),
-        [username],
-    )
+
+    useEffect(() => {
+        activatedSocialNetworkUI.collecting.identityProvider?.recognized.addListener((val) => {
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx]: Need to remove the listener.

File: packages/mask/src/components/InjectedComponents/SetupGuide.tsx [hunk 7]
```diff
@@ -460,51 +401,105 @@ function SetupGuideUI(props: SetupGuideUIProps) {
...
+            // twitter will redirect to home page after login
+            if (!(!reloaded && location.pathname === '/home')) return
+            reloaded = true
+            location.reload()
+        }
+        window.addEventListener('locationchange', handler)
+        return () => {
+            window.removeEventListener('locationchange', handler)
+        }
+    }, [username])
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/SetupGuide.tsx:438]:
```suggestion
    }, [username, location.pathname])
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**